pre-commit.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
```

Comment:
I'm not particularly a fan of cluttering the root of a source tree with extra files, especially when there could (potentially) be more hooks created later. So let's make a .githooks directory and move this file there.

Reply:
fair point.  done.